controllers/reclaimspacejob_controller.go

```diff
@@ -376,6 +376,11 @@ func (r *ReclaimSpaceJobReconciler) nodeReclaimSpace(
 	ctx context.Context,
 	logger *logr.Logger,
 	target *targetDetails) (bool, *int64, error) {
+	if target.nodeID == "" {
```

Comment:
it is cleaner to skip calling this function altogether. do the check where `nodeReclaimSpace()` is called?

Reply:
is this means VA object is created but no NodeID is added to it?  explaining in what case it can happen will be helpful.

Reply:
maybe add a helper `func (td *targetDetails) canNodeReclaimSpace() bool {}` so that the check for `target.nodeID` does not need to be placed in the execution flow of `reconcile()`. You might be able to drop the 1st return value from `nodeReclaimSpace()` that way too?

Reply:
> is this means VA object is created but no NodeID is added to it? explaining in what case it can happen will be helpful.

It means no VA object exists for that pvc, I'll add comments




> maybe add a helper `func (td *targetDetails) canNodeReclaimSpace() bool {}` so that the check for `target.nodeID` does not need to be placed in the execution flow of `reconcile()`. You might be able to drop the 1st return value from `nodeReclaimSpace()` that way too?

okay :+1: